webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/repository/xml/GitArchivers.scala

```diff
@@ -409,28 +424,28 @@ class GitActiveTechniqueCategoryArchiverImpl(
       getParents: List[ActiveTechniqueCategoryId],
       gitCommit:  Option[(ModificationId, PersonIdent, Option[String])]
   ): IOResult[GitPath] = {
-    val uptcFile = newActiveTechniquecFile(uptcId, getParents)
-    val gitPath  = toGitPath(uptcFile)
-    if (uptcFile.exists) {
-      for {
-        // don't forget to delete the category *directory*
-        deleted  <- IOResult.attempt(FileUtils.forceDelete(uptcFile))
-        _        <- logPure.debug("Deleted archived technique library category: " + uptcFile.getPath)
-        commited <- gitCommit match {
-                      case Some((modId, commiter, reason)) =>
-                        commitRmFileWithModId(
-                          modId,
-                          commiter,
-                          gitPath,
-                          s"Delete archive of technique library category with ID '${uptcId.value}'${GET(reason)}"
-                        )
-                      case None                            => ZIO.unit
-                    }
-      } yield {
-        GitPath(gitPath)
-      }
-    } else {
-      GitPath(gitPath).succeed
+    for {
+      uptcFile <- newActiveTechniquecFile(uptcId, getParents)
+      gitPath   = toGitPath(uptcFile)
+      // don't forget to delete the category *directory*
+      _        <- ZIO.whenZIO(IOResult.attempt(uptcFile.exists)) {
```

Comment:
Sometimes I added this `IOResult.attempt` wrapper check over `file.exists` I/O calls...